src/System.Private.CoreLib/shared/System/Runtime/AmbiguousImplementationException.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime;
+
+namespace System.Runtime.Serialization
+{
+    [Serializable]
+    public sealed class AmbiguousImplementationException : Exception
+    {
+        private static string s_nullMessage = SR.AmbiguousImplementationException_NullMessage;
+        private static string s_defaultMessage = SR.AmbiguousImplementationException;
+
+        public AmbiguousImplementationException()
+            : base(s_nullMessage)
+        {
+        }
+
+        public AmbiguousImplementationException(string methodName, string interfaceName, string typeName, string assemblyName)
+            : base(String.Format(CultureInfo.CurrentCulture, s_defaultMessage, methodName, interfaceName, typeName, assemblyName))
+        {
+        }
+
+        public AmbiguousImplementationException(string message)
+            : base(message)
+        {
+        }
+
+        public AmbiguousImplementationException(string message, Exception innerException)
+            : base(message, innerException)
+        {
+        }
+
+        public AmbiguousImplementationException(SerializationInfo info, StreamingContext context)
```

Comment:
It was requested that the class be sealed. I made this constructor public (that was the force-push just now). Should it be private instead?

Reply:
The binary serialization constructors for exceptions should never be public.

Yes, make it private. It is what is done for other sealed exception types.

Reply:
Yes, private is good.

Reply:
Roger dodger.